plan/logical_plan_builder.go

```diff
@@ -1343,10 +1343,9 @@ func (b *planBuilder) buildSemiApply(outerPlan, innerPlan LogicalPlan, condition
 	b.optFlag = b.optFlag | flagBuildKeyInfo
 	b.optFlag = b.optFlag | flagDecorrelate
 	join := b.buildSemiJoin(outerPlan, innerPlan, condition, asScalar, not)
-	ap := &LogicalApply{LogicalJoin: *join}
-	ap.tp = TypeApply
-	ap.id = ap.tp + ap.allocator.allocID()
-	ap.self = ap
+	ap := LogicalApply{
+		LogicalJoin: *join,
```

Comment:
`LogicalJoin` should be assigned after call `init()`

Reply:
we'd better not call `init` here. after `init`, we should `ap.LogicalJoin = *join`, then we have to realloc `id` and reset `tp`